### PR TITLE
Add new option `allArgConstructor` for java client, spring generators

### PR DESCRIPTION
**Pull request by wing328**
based on https://github.com/OpenAPITools/openapi-generator/pull/18405

remove EOL in pojo template

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java [hunk 1]
```diff
@@ -1899,38 +1900,6 @@ public void testMicroprofileGenerateCorrectJsonbCreator_issue12622() throws Exce
                 .containsWithNameAndAttributes("JsonbProperty", ImmutableMap.of("value", "\"c\""));
     }
 
-    @Test
-    public void testMicroprofileGenerateCorrectJacksonGenerator_issue18336() throws Exception {
```
Comment (vanduc2514) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: @wing328 why this test was removed ?

Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: thanks for pointing it out. i missed that during review

@jpfinne can you please take a look?

Comment (jpfinne) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: This test did not exist in 7.5.0.

So most probably a merge issue.

Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: you did remove it in your PR: https://github.com/OpenAPITools/openapi-generator/pull/18405/files#diff-73bab52f5d7424b5d166fb81aa39bfe47a4019ad4b9eb207b05a01c449d0c9b5L1902 so I don't think it's a merge issue.

Comment (jpfinne) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: A manual merge from master to my branch, with a manual handling of the conflicts.

i.e. a mistake

Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: that's ok. it happened before.

can you please file a PR to restore those changes (deleted as part of the merge from master)?

Comment (wing328) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaClientCodegenTest.java:1903]: > Btw: I find it hard to justifiy the time spent on this project when the commit are not done in my name

what do you mean?

the commit correctly shows you as the co-author:

```
commit 2a1b4f90df4974a497fd0fe5a545c09007c12688
Author: William Cheng <wing328hk@gmail.com>
Date:   Wed May 1 18:10:35 2024 +0800

    Add new option `allArgConstructor` for java client, spring generators (#18538)

    * allArgConstructor for java

    * Remove leftover from builder pattern branch

    * Rename generateConstructorWithAllArgs and use x- in vendorExtensions

    * Test issue #18340

    * Add evidences for allVars issue (#18340)

    * remove eol

    * update doc

    ---------

    Co-authored-by: jpfinne <jeanpaul@finne.be>

```

**Pull request merged**